### PR TITLE
refactor: update `blas/ext/base/dsnansum` to follow current project conventions

### LINKED ISSUE #1492: [RFC]: refactor `blas/ext/base/dsnansum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dsnansum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dsnansum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Shubh942**
Resolves #1492.

## Description

update blas/ext/base/dsnansum to follow current project conventions
> What is the purpose of this pull request?

This pull request:

progresses #1152 
Fixes #1492 

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Shubh942** [comment]
@Planeshifter @Pranavchiku @kgryte can you please review it

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2020 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/README.md]:
```suggestion
Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/README.md [hunk 2]
```diff
@@ -139,22 +132,11 @@ var v = dsnansum.ndarray( N, x, 2, 1 );
...
-
-x = new Float32Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-    if ( randu() < 0.2 ) {
-        x[ i ] = NaN;
-    } else {
-        x[ i ] = round( randu()*100.0 );
-    }
-}
+var x = filledarrayBy(10, 'float32', discreteUniform(0, 100));
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.js [hunk 4]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float32Array = require( '@stdlib/array/float32' );
 var pkg = require( './../package.json' ).name;
 var dsnansum = require( './../lib/dsnansum.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.native.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.native.js [hunk 6]
```diff
@@ -36,6 +36,7 @@ var dsnansum = tryRequire( resolve( __dirname, './../lib/dsnansum.native.js' ) )
 var opts = {
 	'skip': ( dsnansum instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.ndarray.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.ndarray.js [hunk 8]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float32Array = require( '@stdlib/array/float32' );
 var pkg = require( './../package.json' ).name;
 var dsnansum = require( './../lib/ndarray.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.ndarray.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.ndarray.native.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.ndarray.native.js [hunk 10]
```diff
@@ -36,6 +36,7 @@ var dsnansum = tryRequire( resolve( __dirname, './../lib/ndarray.native.js' ) );
 var opts = {
 	'skip': ( dsnansum instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.ndarray.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/docs/repl.txt [hunk 11]
```diff
@@ -37,17 +37,13 @@
 
     // Using `N` and `stride` parameters:
     > x = new {{alias:@stdlib/array/float32}}( [ -2.0, 1.0, 1.0, -5.0, 2.0, -1.0, NaN, NaN ] );
-    > var N = {{alias:@stdlib/math/base/special/floor}}( x.length / 2 );
-    > var stride = 2;
-    > {{alias}}( N, x, stride )
+    > {{alias}}( 3, x, 2 )
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/docs/repl.txt]:
```suggestion
    > {{alias}}( 4, x, 2 )
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/docs/repl.txt [hunk 12]
```diff
@@ -37,17 +37,13 @@
...
     1.0
 
     // Using view offsets:
     > var x0 = new {{alias:@stdlib/array/float32}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0, NaN, NaN ] );
     > var x1 = new {{alias:@stdlib/array/float32}}( x0.buffer, x0.BYTES_PER_ELEMENT*1 );
-    > N = {{alias:@stdlib/math/base/special/floor}}( x0.length / 2 );
-    > stride = 2;
-    > {{alias}}( N, x1, stride )
+    > {{alias}}( 3, x1, 2 )
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/docs/repl.txt]:
```suggestion
    > {{alias}}( 4, x1, 2 )
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/examples/index.js [hunk 13]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/examples/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/index.js [hunk 14]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/index.js [hunk 15]
```diff
@@ -35,13 +35,11 @@
...
 * @example
 * var Float32Array = require( '@stdlib/array/float32' );
-* var floor = require( '@stdlib/math/base/special/floor' );
 * var dsnansum = require( '@stdlib/blas/ext/base/dsnansum' );
 *
 * var x = new Float32Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
-* var N = floor( x.length / 2 );
 *
-* var v = dsnansum.ndarray( N, x, 2, 1 );
+* var v = dsnansum.ndarray( 4, x, 2, 1 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/index.js]:
```suggestion
* var v = dsnansum.ndarray( 5, x, 2, 1 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/ndarray.js [hunk 16]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/ndarray.js [hunk 17]
```diff
@@ -36,12 +36,10 @@ var dsnansumpw = require( '@stdlib/blas/ext/base/dsnansumpw' ).ndarray;
 *
 * @example
 * var Float32Array = require( '@stdlib/array/float32' );
-* var floor = require( '@stdlib/math/base/special/floor' );
 *
 * var x = new Float32Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
-* var N = floor( x.length / 2 );
 *
-* var v = dsnansum( N, x, 2, 1 );
+* var v = dsnansum( 8, x, 2, 1 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/ndarray.js]:
```suggestion
* var v = dsnansum( 5, x, 2, 1 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/ndarray.native.js [hunk 18]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/ndarray.native.js [hunk 19]
```diff
@@ -37,20 +38,16 @@ var addon = require( './dsnansum.native.js' );
 *
 * @example
 * var Float32Array = require( '@stdlib/array/float32' );
-* var floor = require( '@stdlib/math/base/special/floor' );
 *
 * var x = new Float32Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0, NaN, NaN ] );
-* var N = floor( x.length / 2 );
 *
-* var v = dsnansum( N, x, 2, 1 );
+* var v = dsnansum( 4, x, 2, 1 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/lib/ndarray.native.js]:
```suggestion
* var v = dsnansum( 5, x, 2, 1 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json [hunk 20]
```diff
@@ -35,7 +35,11 @@
 			],
 			"libpath": [],
 			"dependencies": [
-				"@stdlib/blas/ext/base/dsnansumpw"
+				"@stdlib/blas/ext/base/dsnansumpw",
+				"stdlib/napi/export.h",
+				"stdlib/napi/argv.h",
+				"stdlib/napi/argv_int64.h",
+				"stdlib/napi/argv_strided_float32array.h"
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json]: This file has TAB indentation, please fix this.

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js [hunk 21]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js [hunk 22]
```diff
@@ -142,18 +141,18 @@ tape( 'the function supports a `stride` parameter', function test( t ) {
 		NaN
 	]);
 
-	N = floor( x.length / 2 );
+	N = 4;
 	v = dsnansum( N, x, 2 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js]:
```suggestion
	v = dsnansum( 4, x, 2 );
```

Apply this in every `test.*` file

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js [hunk 23]
```diff
@@ -168,7 +167,7 @@ tape( 'the function supports a negative `stride` parameter', function test( t )
 		2.0
 	]);
 
-	N = floor( x.length / 2 );
+	N = 4;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js]: same comment here

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js [hunk 24]
```diff
@@ -215,8 +214,8 @@ tape( 'the function supports view offsets', function test( t ) {
 		NaN
 	]);
 
+	N = 4;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js]: same comment here, also remove declaration of `N`

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
You have to fix `addon.c` file, refer similar PRs. Left few comments, once addressed this will be good to merge.

**Comment by Shubh942** [comment]
I will change the code as specified. Thanks for your response

**Comment by Shubh942** [comment]
@Pranavchiku I have made the suggested changes.

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json [hunk 25]
```diff
@@ -1,42 +1,41 @@
...
+		"libraries": ["-lm"],
+		"libpath": [],
+		"dependencies": [
+		  "@stdlib/blas/ext/base/dsnansumpw",
+		  "stdlib/napi/export.h",
+		  "stdlib/napi/argv.h",
+		  "stdlib/napi/argv_int64.h",
+		  "stdlib/napi/argv_strided_float32array.h"
+		]
+	  }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json]: This is still tab indented, please refer similar PRs and update accordingly

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
You will have to update indentation in `manifest.json` and update `addon.c`

**Comment by Shubh942** [comment]
@Pranavchiku I have updated the indentation in `manifest.json` and update `addon.c`.

**Comment by Shubh942** [comment]
@Planeshifter can you please review my PR

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json [hunk 26]
```diff
@@ -1,42 +1,77 @@
...
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": []
+    }
+  ]
 }
+
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json]:
```suggestion
}

```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c [hunk 27]
```diff
@@ -0,0 +1,47 @@
...
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, sum, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, Y, N, strideY, argv, 4 );
+	stdlib_strided_dsnansum( N, sum, X, strideX, Y, strideY );
+	return NULL;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );

	napi_value v;
	status = napi_create_double( env, stdlib_strided_dnansum( N, X, stride ), &v );
	assert( status == napi_ok );
	return v;
}
```

**Review by Pranavchiku — APPROVED** [review]
This looks good to me. Thanks @Shubh942 !

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c [hunk 28]
```diff
@@ -0,0 +1,48 @@
...
+*
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 2 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c:38]:
```suggestion
	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 2 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c [hunk 29]
```diff
@@ -0,0 +1,48 @@
...
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 1 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 1 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c (hunk 28 above)
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 2 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c [hunk 30]
```diff
@@ -0,0 +1,48 @@
...
+
+	napi_value v;
+	status = napi_create_double( env, stdlib_strided_dnansum( N, X, stride ), &v );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c]:
```suggestion
	status = napi_create_double( env, stdlib_strided_dsnansum( N, X, stride ), &v );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json [hunk 31]
```diff
@@ -1,42 +1,76 @@
...
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv",
+        "@stdlib/napi/argv-float",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json]:
```suggestion
```

This isn't needed.

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c [hunk 32]
```diff
@@ -0,0 +1,47 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/dsnansum.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_float.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/src/addon.c]:
```suggestion
```

This isn't used.

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.ndarray.js [hunk 33]
```diff
@@ -35,7 +34,7 @@ tape( 'main export is a function', function test( t ) {
 });
 
 tape( 'the function has an arity of 4', function test( t ) {
-	t.strictEqual( dsnansum.length, 4, 'has expected arity' );
+	t.strictEqual( dsnansum.length, 4, 'return expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.ndarray.js]:
```suggestion
	t.strictEqual( dsnansum.length, 4, 'returns expected value' );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js [hunk 34]
```diff
@@ -35,7 +34,7 @@ tape( 'main export is a function', function test( t ) {
 });
 
 tape( 'the function has an arity of 3', function test( t ) {
-	t.strictEqual( dsnansum.length, 3, 'has expected arity' );
+	t.strictEqual( dsnansum.length, 3, 'return expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.dsnansum.js]:
```suggestion
	t.strictEqual( dsnansum.length, 3, 'returns expected value' );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.ndarray.native.js [hunk 35]
```diff
@@ -44,7 +43,7 @@ tape( 'main export is a function', opts, function test( t ) {
 });
 
 tape( 'the function has an arity of 4', opts, function test( t ) {
-	t.strictEqual( dsnansum.length, 4, 'has expected arity' );
+	t.strictEqual( dsnansum.length, 4, 'return expected value' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/test/test.ndarray.native.js]:
```suggestion
	t.strictEqual( dsnansum.length, 4, 'returns expected value' );
```

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/README.md [hunk 36]
```diff
@@ -139,22 +131,11 @@ var v = dsnansum.ndarray( N, x, 2, 1 );
...
-
-x = new Float32Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-    if ( randu() < 0.2 ) {
-        x[ i ] = NaN;
-    } else {
-        x[ i ] = round( randu()*100.0 );
-    }
-}
+var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/README.md]: The changes to this example are not equivalent to the original.

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.native.js [hunk 37]
```diff
@@ -48,17 +49,7 @@ var opts = {
...
-
-	x = new Float32Array( len );
-	for ( i = 0; i < x.length; i++ ) {
-		if ( randu() < 0.2 ) {
-			x[ i ] = NaN;
-		} else {
-			x[ i ] = ( randu()*10.0 ) - 20.0;
-		}
-	}
+	var x = filledarrayBy( len, 'float32', rand );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/benchmark/benchmark.native.js:52]: The changes to this (and the other benchmarks) are not equivalent to the original.

File: lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json [hunk 38]
```diff
@@ -1,42 +1,76 @@
...
-			],
-			"include": [
-				"./include"
-			],
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": [
-				"@stdlib/blas/ext/base/dsnansumpw"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dsnansum/manifest.json:38]: This dependency has been omitted from the dependencies in each of the configurations. This should be re-added to each of them.

**Review by kgryte — CHANGES_REQUESTED** [review]
This PR needs changes, as directed in the comments, before it can be merged. Benchmark test data generation should remain similar to the original. Namely, `NaNs` should be present; otherwise, compiler optimizations can result in conditionals being removed, resulting in better measured performance than "real-world" usage.

**Comment by Shubh942** [comment]
@kgryte @Planeshifter I made the required changes. Can you please review it.

**Comment by Shubh942** [comment]
@kgryte  @Planeshifter I made the required changes.

**Comment by kgryte** [comment]
@Shubh942 Looks like this PR is failing CI, including failing tests and lint errors. Until those are resolved, we won't be able to move this PR forward.

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**